src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java

```diff
@@ -2,120 +2,110 @@
 import com.fasterxml.jackson.databind.module.SimpleModule;
-import de.azapps.kafkabackup.common.record.Record;
 
 import static org.junit.Assert.assertEquals;
 
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 
-import java.io.*;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.nio.file.Files;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
 import java.util.Base64;
-import java.util.Comparator;
 
 public class RecordJSONSerdeTest {
 
+    private static final String topic = "test-topic";
+    private static final int partition = 42;
+    private static final long offset = 123;
+    private static byte[] keyBytes;
+    private static String keyBase64;
+    private static byte[] valueBytes;
+    private static String valueBase64;
+    private static final String JSON_ENCODING = "UTF-8";
+    private static ObjectMapper mapper;
+    private static RecordJSONSerde serde;
+
+    @BeforeClass
+    public static void beforeAll() throws Exception {
+        // encoding here is not really important, we just want some bytes
+        keyBytes = "test-key".getBytes("UTF-8");
+        valueBytes = "test-value".getBytes("UTF-8");
+        // using Base64 as encoding in the json is part of our Record serialization format however:
+        keyBase64 = Base64.getEncoder().encodeToString(keyBytes);
+        valueBase64 = Base64.getEncoder().encodeToString(valueBytes);
+    }
+
+    @Before
+    public void beforeEach() throws Exception {
+        mapper = new ObjectMapper();
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(Record.class, new RecordJSONSerde.Deserializer());
+        module.addSerializer(Record.class, new RecordJSONSerde.Serializer());
+        mapper.registerModule(module);
+
+        serde = new RecordJSONSerde();
+    }
+
     @Test
     public void readTest() throws Exception {
         // GIVEN
-        String topic = "test-topic";
-        int partition = 42;
-        long offset = 123;
-        // encoding here is not really important, we just want some bytes
-        byte[] key = "test-key".getBytes("UTF-8");
-        byte[] value = "test-value".getBytes("UTF-8");
-        // using Base64 as encoding here is part of our Record serialization format however:
-        String keyBase64 = Base64.getEncoder().encodeToString(key);
-        String valueBase64 = Base64.getEncoder().encodeToString(value);
         // TODO: add timestamp, timestampType, and headers
-        byte[] json = String.format("{\"topic\":\"%s\",\"partition\":%d,\"offset\":%d,\"key\":\"%s\",\"value\":\"%s\"}", topic, partition, offset, keyBase64, valueBase64).getBytes("UTF-8");
+        byte[] json = String.format("{\"topic\":\"%s\",\"partition\":%d,\"offset\":%d,\"key\":\"%s\",\"value\":\"%s\"}", topic, partition, offset, keyBase64, valueBase64).getBytes(JSON_ENCODING);
```

Comment:
maybe can we add some "test.json" file and read value from this file? what do you think?

Reply:
I tried to add a helper, but then I realised that the json is different in almost every test, especially when testing different edge-cases like missing fields, null fields etc... But I don't like this super long line of weird formatting either... 😏

Reply:
so the other idea is to extract this to separated file(ex. TestJsonFiles, TestJsonRecord etc.)with static fields like `JSON_WITH_MISSING_FIELD` and then we'll avoid such long lines.

Reply:
Fixed this in a follow up PR